### DISCUSSION TITLE
CRS container with incomplete information

### DISCUSSION THREAD

**Discussion by djhoese** [pyproj4/pyproj#778] [Ideas]
One thing we in the pyresample/satpy community sometimes use is the idea of a partial CRS where we will fill in some of the parameters later. We do this currently with PROJ parameter dictionaries. One use case would be: I am going to resample my data to a target projection/grid, but I don't know all of the parameters yet. So maybe I know I want a LCC projection and I want `lat_1` and `lat_2` to be `25.0`, but I want to center the projection (`lon_0`, longitude of natural origin) to the middle of the data.

So in the code I could do something like:

```
partial_dict = {'proj': 'lcc', 'lat_1': 25.0, 'lat_2': 25.0, 'datum': 'WGS84'}

# ...various interfaces and function calls and then...

lons, lats = get_data_lonlats()
middle_lon = (lons.max() - lons.min()) / 2  # I know this wouldn't work, just an example
partial_dict['lon_0'] = middle_lon

from pyproj import CRS
crs = CRS(partial_dict)

# do resampling
```

Is there any way to do something like this with pyproj **without** going back to PROJ dicts? Is there any functionality for assigning/replacing parameters on a CRS object?

**Comment by snowman2** [comment]
CRS objects are immutable. But, you can achieve what you want with the functionality to build a CRS: https://pyproj4.github.io/pyproj/stable/build_crs.html

**Reply by djhoese** [reply]
Ah that's an interesting idea. This would require keeping track of the kwargs for the coordinate operation and hold on to all the other components (ellipsoid, datum, etc). This could definitely be a path forward though.
